src/core/vanilla.ts

```diff
@@ -161,7 +161,7 @@ const setAtomReadError = <Value>(
   state: State,
   atom: Atom<Value>,
   error: Error,
-  dependencies: Set<AnyAtom>,
+  dependencies?: Set<AnyAtom>,
```

Comment:
Isn't it better to send in an empty set as argument, when there's no dependencies?

I'm scared we'll forget to parse it at a later point.

Reply:
Unfortunately, it can't be an empty set. It's different.
This is already the pattern used everywhere, not just in this PR change.
Let's discuss separately for better readability, when we get time.